cli/wordpress-server-child.ts

```diff
@@ -0,0 +1,187 @@
+const originalStderrWrite = process.stderr.write.bind( process.stderr );
+
+process.stdout.write = function ( ...args: Parameters< typeof originalStdoutWrite > ) {
+	if ( process.send ) {
+		process.send( { type: 'activity' } );
+	}
+	return originalStdoutWrite( ...args );
+} as typeof process.stdout.write;
+
+process.stderr.write = function ( ...args: Parameters< typeof originalStderrWrite > ) {
+	if ( process.send ) {
+		process.send( { type: 'activity' } );
+	}
+	return originalStderrWrite( ...args );
+} as typeof process.stderr.write;
+
+function escapePhpString( str: string ): string {
+	return str.replace( /\\/g, '\\\\' ).replace( /'/g, "\\'" );
+}
+
+async function setAdminPassword( server: RunCLIServer, adminPassword: string ): Promise< void > {
+	await server.playground.request( {
+		url: '/?studio-admin-api',
+		method: 'POST',
+		body: {
+			action: 'set_admin_password',
+			password: escapePhpString( adminPassword ),
+		},
+	} );
+}
+
+async function startServer( config: ServerConfig ): Promise< void > {
+	if ( server ) {
+		console.log( `[WordPress Server Child] Server already running for site ${ config.siteId }` );
+		return;
+	}
+
+	try {
+		const hasWordPress = isWordPressDirectory( config.sitePath );
+
+		const [ studioMuPluginsHostPath, loaderMuPluginHostPath ] = await getMuPlugins( {
+			isWpAutoUpdating: config.isWpAutoUpdating,
+		} );
+
+		const defaultConstants = {
+			WP_SQLITE_AST_DRIVER: true,
+		};
+
+		const mounts = [
+			{
+				hostPath: config.sitePath,
+				vfsPath: '/wordpress',
+			},
+			{
+				hostPath: studioMuPluginsHostPath,
+				vfsPath: '/internal/studio/mu-plugins',
+			},
+			{
+				hostPath: loaderMuPluginHostPath,
+				vfsPath: '/internal/shared/mu-plugins/99-studio-loader.php',
+			},
+		];
+
+		const args: RunCLIArgs = {
+			command: 'server',
+			internalCookieStore: false,
+			login: false,
+			followSymlinks: true,
+			skipSqliteSetup: true,
+			port: config.port,
+			'mount-before-install': mounts,
+			'site-url': config.absoluteUrl || `http://localhost:${ config.port }`,
+			blueprint: config.blueprint || {},
+			wordpressInstallMode: 'download-and-install',
+		};
+
+		if ( hasWordPress ) {
+			args.wordpressInstallMode = 'install-from-existing-files-if-needed';
+		}
+
+		if ( config.phpVersion ) {
+			args.php = config.phpVersion as SupportedPHPVersion;
+		}
+
+		if ( config.wpVersion ) {
+			if ( isWordPressDevVersion( config.wpVersion ) ) {
+				args.wp = 'nightly';
+			} else {
+				args.wp = config.wpVersion;
+			}
+		}
+
+		args.blueprint.constants = { ...args.blueprint.constants, ...defaultConstants };
+		const server = await runCLI( args );
+
+		if ( ! server ) {
+			throw new Error( 'Failed to start server: runCLI returned void' );
+		}
+
+		if ( config.adminPassword ) {
+			await setAdminPassword( server, config.adminPassword );
+		}
+	} catch ( error ) {
+		server = null;
+		console.error( `[WordPress Server Child] Failed to start server:`, error );
+		throw error;
+	}
```

Comment:
It seems to me that we should explicitly kill the process if this happens, no? Maybe we should send an `error` response first, but leaving the process hanging seems like a bad idea.